/*
 * Copyright (c) 2024 Greg Padiasek
 * Distributed under the terms of the MIT License.
 * See accompanying file LICENSE or copy at http://opensource.org/licenses/MIT
 */

//
// Seegnify graph messages
//

syntax = "proto2";
package seegnify.graph;

// GetWeights request

message GetWeights {
  optional string version = 1;
  required uint64 position = 2;
}

// GetWeights response

message GetWeightsResponse {
  required string version = 1;
  required bytes buffer = 2;
  required bool complete = 3;
}

// SetWeights request

message SetWeights {
  optional string version = 1;
  required bytes buffer = 2;
  required bool complete = 3;
}

// SetWeights response

message SetWeightsResponse {
  required string version = 1;
}

// Update Weights request

message UpdWeights {
  optional string version = 1;
  required bytes buffer = 2;
  required bool complete = 3;
}

// Update Weights response

message UpdWeightsResponse {
  required string version = 1;
}

// Success response

message SuccessResponse {
}

// Failure response

message ErrorResponse {
  required uint32 status = 1;  
  required string message = 2;
}

// Message envelopes (union)

message Request {
  oneof request {
    seegnify.graph.GetWeights get_weights = 10;
    seegnify.graph.SetWeights set_weights = 11;
    seegnify.graph.UpdWeights upd_weights = 12;
  }
}

message Response {
  oneof response {
    seegnify.graph.GetWeightsResponse get_weights = 11;
    seegnify.graph.SetWeightsResponse set_weights = 12;
    seegnify.graph.UpdWeightsResponse upd_weights = 13;
    seegnify.graph.SuccessResponse success = 14;
    seegnify.graph.ErrorResponse error = 15;
  }
}

